File: nodecore-spv/src/main/java/veriblock/validator/LedgerProofReplyValidator.java
package veriblock.validator;

import nodecore.api.grpc.VeriBlockMessages;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.veriblock.extensions.ledger.LedgerProofWithContext;
import veriblock.net.impl.PeerTableImpl;

public class LedgerProofReplyValidator {
    private static final Logger logger = LoggerFactory.getLogger(PeerTableImpl.class);

    public static boolean validate(VeriBlockMessages.LedgerProofReply.LedgerProofResult ledgerProofResult){
        try {
            LedgerProofWithContext ledgerProofWithContext = LedgerProofWithContext.parseFrom(ledgerProofResult.getLedgerProofWithContext());
        } catch (Exception ex){
            logger.warn("LedgerProofWithContext is not valid. " + ex.getMessage(), ex);
            //TODO SPV-67 add ban for peer.
            return false;
        }
        return true;
    }


}

File: nodecore-spv/src/main/java/veriblock/validator/LedgerProofReplyValidator.kt
package veriblock.validator

import nodecore.api.grpc.VeriBlockMessages.LedgerProofReply.LedgerProofResult
import org.slf4j.LoggerFactory
import org.veriblock.core.utilities.createLogger
import org.veriblock.extensions.ledger.LedgerProofWithContext
import veriblock.net.SpvPeerTable

private val logger = createLogger {}

object LedgerProofReplyValidator {
    @JvmStatic
    fun validate(ledgerProofResult: LedgerProofResult): Boolean {
        try {
            LedgerProofWithContext.parseFrom(
                ledgerProofResult.ledgerProofWithContext
            )
        } catch (ex: Exception) {
            logger.warn(ex) {"LedgerProofWithContext is not valid. ${ex.message}" }
            //TODO SPV-67 add ban for peer.
            return false
        }
        return true
    }
}
